src/components/carousel/index.tsx

```diff
@@ -1,87 +1,102 @@
-import React, {
-  FC,
-  PropsWithChildren,
-  ReactNode,
-  useCallback,
-  useEffect,
-  useState,
-} from 'react';
+import React, { FC, ReactNode, useCallback, useEffect, useState } from 'react';
 import useEmblaCarousel from 'embla-carousel-react';
-import { useMultiStyleConfig } from '@chakra-ui/react';
+import { useMediaQuery, useMultiStyleConfig } from '@chakra-ui/react';
 
+import ThumbnailPagination from './ThumbnailPagination';
 import Slide from './Slide';
-import NavigationButton, { Direction } from './NavigationButton';
+import NavigationButton from './NavigationButton';
 import Flex from '../flex';
 import Box from '../box';
+import { breakpoints } from '../../themes';
 
-interface Props {
+type SharedProps = {
   startIndex?: number;
   onSlideClick?: (index: number) => void;
   onSlideSelect?: (index: number) => void;
-  fullScreen?: boolean;
-}
-
-const Carousel: FC<PropsWithChildren<Props>> = ({
-  startIndex = 0,
-  onSlideClick,
-  onSlideSelect,
-  children,
-  fullScreen = false,
-}) => {
-  const { container, slideContainer } = useMultiStyleConfig(
+};
+
+type DefaultProps = {
+  fullScreen?: never;
+  children: ReactNode[];
+} & SharedProps;
+
+type FullScreenSlide = { slide: ReactNode; thumbnail: ReactNode };
+type FullScreenProps = {
+  fullScreen: true;
+  children: Array<FullScreenSlide>;
```

Comment:
It feels a bit wired to have the childer of an FC being polymorphic. But the only other thing I could think of straight away is to extract the image form the passed slide instead

Reply:
The carousel should not only allow images but also other content like cards. That is why @Averethel and I decided to use children for it. The reason we went for an array of objects is that you want to load different-sized images and video for the thumbnails.